### PR TITLE
convert diffusion_test setup to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
Running this in 2-d axisymmetry, I get:

```
 nx         error
 32         0.001479796186
 64         0.0003684763597
128        9.406415352e-05
````
this is 2nd order convergence as expected

**Comment by zingale** [comment]
test: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-11-22-008/index.html

File: Exec/unit_tests/diffusion_test/Problem_Derive.cpp [hunk 1]
```diff
@@ -13,52 +14,31 @@ void deranalytic(const Box& bx, FArrayBox& derfab, int dcomp, int /*ncomp*/,
                  Real time, const int* /*bcrec*/, int /*level*/)
 {
 
-  // derive the dynamic pressure
+    // derive the dynamic pressure
```
Comment (maxpkatz) [inline Exec/unit_tests/diffusion_test/Problem_Derive.cpp:17]: We should fix this comment

**Review by maxpkatz — APPROVED** [review]

File: Exec/unit_tests/diffusion_test/Problem_Derive.cpp (hunk 1 above)
Comment (zingale) [reply Exec/unit_tests/diffusion_test/Problem_Derive.cpp]: fixed

**Pull request merged**